### PR TITLE
Parser for AWS Inspector2 findings

### PR DESCRIPTION
**Pull request by siniysv**
A parser for AWS Inspector2 ("new" Inspector) findings exported with `aws inspector2 list-findings` in JSON format.

**Checklist**

This checklist is for your information.

- [x] Make sure to rebase your PR against the very latest `dev`.
- [x] Features/Changes should be submitted against the `dev`.
- [x] Bugfixes should be submitted against the `bugfix` branch.
- [x] Give a meaningful name to your PR, as it may end up being used in the release notes.
- [x] Your code is flake8 compliant.
- [x] Your code is python 3.11 compliant.
- [x] If this is a new feature and not a bug fix, you've included the proper documentation in the docs at https://github.com/DefectDojo/django-DefectDojo/tree/dev/docs as part of this PR.
- [x] Model changes must include the necessary migrations in the dojo/db_migrations folder.
- [x] Add applicable tests to the unit tests.
- [x] Add the proper label to categorize your PR.

[sc-8014]

### CODE REVIEW

File: dojo/tools/aws_inspector2/parser.py
Comment (manuel-sommer) [inline dojo/tools/aws_inspector2/parser.py:7]: A lot of AWS users use AWS Security Hub to retrieve all findings. Could you also advance this parser: https://github.com/DefectDojo/django-DefectDojo/blob/master/dojo/tools/awssecurityhub/inspector.py ?

Are the returned AWS Security Hub findings of Inspector2 equal to the returned findings of Inspector2 or are they differently formatted?

File: dojo/settings/settings.dist.py
Comment (manuel-sommer) [inline dojo/settings/settings.dist.py:1276]: Either add more fields as the title is probably not unique enough, or only use uniqueidfromtool. Furthermore, you forgot to edit the settings.dist.py hash sum https://github.com/DefectDojo/django-DefectDojo/blob/master/dojo/settings/.settings.dist.py.sha256sum

**Review by manuel-sommer — CHANGES_REQUESTED** [review]

File: dojo/tools/aws_inspector2/parser.py
Comment (mtesauro) [reply dojo/tools/aws_inspector2/parser.py:7]: It's definitely preferred to have a single parser that understands multiple formats vs multiple parsers - if nothing else, it's easier for people to use who might not know the difference between Inspector JSON versions.

Comment (siniysv) [reply dojo/tools/aws_inspector2/parser.py:7]: thank you for your attention, I'll try to answer below.
Security Hub collects findings from inspector and a few other AWS services, then it converts the findings into one common format (ASFF - AWS Security Finding Format https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-findings-format.html) and provides API to retrieve the findings in that format. AWS Inspector provides own API with own finding format. It is not necessary to use Security Hub in order to use Inspector since those are 2 different AWS services. Answering to your @manuel-sommer  question - yes, they are differently formatted.

I would extend the existing parser If I had a way to easily detect what finding format is used in an uploaded json, although the name of the parser would be misleading since AWS Security Hub and AWS Inspector are two different services. 

If you believe DefectDojo needs only Security Hub parser because it is indeed an AWS service that can collect findings from other services (full list here https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-internal-providers.html), and there is no need to support standalone AWS service report import, I'm totally okay with this approach. Thank you.

Comment (mtesauro) [reply dojo/tools/aws_inspector2/parser.py:7]: Understood about AWS Security Hub vs AWS Inspector.

I would make sense in this case to have a parser just for AWS Inspector not that you've provided more details - TBH, I can't keep up with all the AWS whatever services.

You will need get the ruff linter happy (as well as the other tests) before the PR will be reviewed and merged (assuming we get 4 approvals).

I'm going to re-kick the tests now - if you want to do that yourself, you can close & open the PR to get them to restart assuming this isn't your first PR against DefectDojo.

File: dojo/fixtures/test_type.json
Comment (Maffooch) [inline dojo/fixtures/test_type.json:56]:
```suggestion
```
This file is no longer modified when adding new parsers

File: dojo/tools/aws_inspector2/parser.py
Comment (Maffooch) [inline dojo/tools/aws_inspector2/parser.py:186]: It would make the code much easier to read by converting individual if cases into a function to process each resource type. This is

**Review by Maffooch — CHANGES_REQUESTED** [review]

**Review by grendel513 — APPROVED** [review]

**Review by Maffooch — APPROVED** [review]

**Review by mtesauro — APPROVED** [review]
Approved

File: dojo/settings/settings.dist.py
Comment (manuel-sommer) [inline dojo/settings/settings.dist.py:1286]: Hi @Maffooch ,
I am not sure if title and severity is sufficient here as Inspector could scan multiple (different) container with the same CVE. Thus, I would recommend to also use the description or information provided in "resources". (e.g. "imageHash": "sha256:c667ad018b538adadb5bd1db75836c8561b23bc7d31d9cd8941fa74f5df93802",)

**Review by manuel-sommer — CHANGES_REQUESTED** [review]

File: dojo/settings/settings.dist.py
Comment (Maffooch) [reply dojo/settings/settings.dist.py:1286]: I can do `title, severity, description`. In most cases, the `unique_id_from_tool` will be used instead

**Review by hblankenship — APPROVED** [review]

**Pull request merged**
